mlir/lib/Dialect/Math/Transforms/ExpandPatterns.cpp

```diff
@@ -615,6 +615,24 @@ static LogicalResult convertRoundEvenOp(math::RoundEvenOp op,
   return success();
 }
 
+// Convert `math.rsqrt` into `arith.divf` + `math.sqrt`
+static LogicalResult convertRsqrtOp(math::RsqrtOp op,
+                                    PatternRewriter &rewriter) {
+
+  auto operand = op.getOperand();
+  auto operandTy = operand.getType();
+  auto eTy = getElementTypeOrSelf(operandTy);
+  if (!isa<FloatType>(eTy))
```

Comment:
I think the expansion will work fine for f16, f32, f64. You can enable them.

Reply:
Sure -- @srcarroll, you're right, F16 and F64 are indeed supported, let me add tests for them.